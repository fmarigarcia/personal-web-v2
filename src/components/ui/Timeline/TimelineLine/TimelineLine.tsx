import React from 'react';

export const TimelineLine: React.FC = () => {
    return (
        <div className="absolute top-16 left-16 right-16 h-0.5 bg-stone-300 dark:bg-zinc-700" />
    );
};
