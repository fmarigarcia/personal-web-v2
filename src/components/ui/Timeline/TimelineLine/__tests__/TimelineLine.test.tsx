import { render } from '@testing-library/react';
import { TimelineLine } from '../TimelineLine';

describe('TimelineLine', () => {
    it('should render with correct classes', () => {
        const { container } = render(<TimelineLine />);

        const line = container.firstChild as HTMLElement;
        expect(line).toHaveClass(
            'absolute',
            'top-16',
            'left-16',
            'right-16',
            'h-0.5',
            'bg-stone-300'
        );
    });

    it('should be a div element', () => {
        const { container } = render(<TimelineLine />);

        const line = container.firstChild as HTMLElement;
        expect(line.tagName).toBe('DIV');
    });

    it('should match snapshot', () => {
        const { container } = render(<TimelineLine />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
