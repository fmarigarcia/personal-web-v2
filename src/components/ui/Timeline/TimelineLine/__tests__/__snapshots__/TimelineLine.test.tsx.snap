// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineLine should match snapshot 1`] = `
<div
  class="absolute top-16 left-16 right-16 h-0.5 bg-stone-300 dark:bg-zinc-700"
/>
`;
